File: Data Structures & Algorithms/generate-parentheses/submission-1.cpp
class Solution {
public:
    void backtrack(string current, int close, int open, int n, vector<string>& ret){
        if (close + open == 2*n){
            ret.push_back(current);
            return;
        }if (open < n){
            backtrack(current + "(", close, open + 1, n, ret);
        }if (close < open){
            backtrack(current + ")", close + 1, open, n, ret);
        }
    }

    vector<string> generateParenthesis(int n) {
        vector<string> ret;
        backtrack("", 0, 0, n, ret);
        return ret;
    }
};

File: Data Structures & Algorithms/longest-palindromic-substring/submission-1.cs
public class Solution {
    public string LongestPalindrome(string s) {
        string ret = "";
        for (int i = 0 ; i < s.Length ; i++){
            int l = i, r = i;
            while(l >= 0 && r < s.Length && s[l] == s[r]){
                if (ret.Length < r - l + 1){
                    ret = s.Substring(l, r - l + 1);
                }r++;l--;
            }l = i ;
            r = i + 1;
            while(l >= 0 && r < s.Length && s[l] == s[r]){
                if (ret.Length < r - l + 1){
                    ret = s.Substring(l , r - l + 1);
                }r++;l--;
            }
        }return ret;
    }
}